chore: move links to h3 new org (#3354)
nitrojs/nitro: docs/1.guide/2.routing.md

#### Single param

> To define a route with params, use the `[<param>]` syntax where `<param>` is the name of the param. The param will be available in the `event.context.params` object or using the `getRouterParam` utility from [h3js/h3](https://h3.dev/).

Reviewer: ```suggestion
To define a route with params, use the `[<param>]` syntax where `<param>` is the name of the param. The param will be available in the `event.context.params` object or using the `getRouterParam` utility from [h3](https://h3.dev/).
```

> To define a route with params, use the `[<param>]` syntax where `<param>` is the name of the param. The param will be available in the `event.context.params` object or using the `getRouterParam` utility from [h3js/h3](https://h3.unjs.io).

Reviewer 2: ```suggestion
To define a route with params, use the `[<param>]` syntax where `<param>` is the name of the param. The param will be available in the `event.context.params` object or using the `getRouterParam` utility from [h3js/h3](https://v1.h3.dev).
```

Reviewer: this PR is targetting nitro v3 which should point to new h3.dev domain

Author: @pi0 , given. :blush:

Nitro allows you to add logic at the top-level for each route of your configuration. It can be used for redirecting, proxying, caching and adding headers to routes.

> It is a map from route pattern (following [unjs/radix3](https://github.com/h3js/rou3/tree/radix3#route-matcher)) to route options.

Reviewer: ```suggestion
It is a map from route pattern (following [rou3](https://github.com/h3js/rou3)) to route options.
```
